common/lib/xmodule/xmodule/js/src/video/03_video_player.js

```diff
@@ -530,6 +535,19 @@ function(HTML5Video, HTML5HLSVideo, Resizer, HLS, _, Time) {
             this.el.trigger('pause', arguments);
         }
         this.el.trigger('ended', arguments);
+        if (config.enableNextOnCompletion === true && config.isComplete === false) {
```

Comment:
why not move this logic inside `handleEnded`
https://github.com/edx/edx-platform/blob/1e4daf749cba3f469423bf08fd4c202d87fba17a/common/lib/xmodule/xmodule/js/src/video/09_completion.js#L103

Reply:
**handleEnded** function is not called on video completion.

Reply:
@ziafazal I and @naumanshafi have debugged this again and watched the whole video. `handleEnded` or `markCompletion` never called during the whole video flow.